controllers/azurecluster_reconciler.go

```diff
@@ -115,11 +118,16 @@ func newAzureClusterService(scope *scope.ClusterScope) (*azureClusterService, er
 			privateEndpointsSvc,
 		},
 		skuCache: skuCache,
-	}, nil
+	}
+	acs.Reconcile = acs.reconcile
+	acs.Pause = acs.pause
+	acs.Delete = acs.delete
+
+	return acs, nil
 }
 
 // Reconcile reconciles all the services in a predetermined order.
-func (s *azureClusterService) Reconcile(ctx context.Context) error {
+func (s *azureClusterService) reconcile(ctx context.Context) error {
```

Comment:
Just thought of this. Why are these functions being changed to lowercase? This would cause the [async controller](https://github.com/willie-yao/cluster-api-provider-azure/blob/clusterclass-aks-final/controllers/azurecluster_controller.go#L223) to not call these functions.
/lgtm remove

Reply:
So these private functions are being referenced here https://github.com/aniruddha2000/cluster-api-provider-azure/blob/ani/issues/4114/controllers/azurecluster_reconciler.go#L122-L124 just like AzureMachineReconciler and being called like this here https://github.com/aniruddha2000/cluster-api-provider-azure/blob/ani/issues/4114/controllers/azurecluster_controller.go#L221

Reply:
Ah that's correct. Thanks for the clarification!
/lgtm

cc @CecileRobertMichon @jackfrancis @mboersma for an approval